### ISSUE TITLE
Record Tour results in "Cannot read property 'getRepository' of undefined"

### ISSUE THREAD

**Issue by tmerkel** [vsls-contrib/codetour#46]
Install extension
Open Command Palette
Choose Record Tour
Get error toast saying: "Cannot read property 'getRepository' of undefined"
Nothing else happens

![image](https://user-images.githubusercontent.com/176968/78174572-51dca400-740e-11ea-89a7-a4b8be9082b4.png)

**Comment by lostintangent** [comment]
@tmerkel Hey! Apologies that you ran into this issue. Few follow-up questions to help me repro:

1. Did you have a workspace open when you tried to record?
1. If so, was the opened workspace also a git repo?
1. Were you running this using VS Code's WSL support?

**Comment by tmerkel** [comment]
No problem!  I love this project so I hope I’m helpful with something :)

Initially had a workspace open.  The workspace contained 4 folders, all separate repos.  Then tried with just a single folder containing a repo and the same issue.

I didn’t try at all with a non-git repo.

I WAS running WSL.  I can try switching over to powershell when I get back home, if that’s helpful?

On Apr 1, 2020, at 12:07 PM, Jonathan Carter <notifications@github.com> wrote:

﻿

@tmerkel<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ftmerkel&data=02%7C01%7C%7C1418840037e544142b6008d7d66ffcae%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637213648771668746&sdata=PS%2BT9vcrr%2FLJ5JAD1%2FKGr6cF5ce0onHqi7JFb3q3%2FaQ%3D&reserved=0> Hey! Apologies that you ran into this issue. Few follow-up questions to help me repro:

  1.  Did you have a workspace open when you tried to record?
  2.  If so, was the opened workspace also a git repo?
  3.  Were you running this using VS Code's WSL support?

**Comment by lostintangent** [comment]
@tmerkel I just confirmed that tour recording should work fine, regardless if the currently opened workspace is a git repo or not. That said, I think there might be an issue with trying to run CodeTour, when VS Code is launched from WSL.

Could you try installing the CodeTour extension in the remote WSL context as opposed to being installed on the "UI" side? I'm wondering if that would fix it.

**Comment by tmerkel** [comment]
Ok...so...  I opened remote, installed there and opened a single repo.  Recording started, no problem.

I then went back to non-remote, reinstalled the extension, opened the folder with the tour... playback and record both work.

They also work when I open up a workspace. 

So...there's something about running in a remote context the first time...or...maybe if I had just had a tour to open first?

Either way...things work now.  Is there something else I could send that would be useful?
